rbac/common/role/role_manager.py

```diff
@@ -14,101 +14,176 @@
+        return message_proto(
+            proposal_id=uuid4().hex,
+            role_id=role_id,
+            user_id=user_id,
+            reason=reason,
+            metadata=metadata,
+        )
+
+    def make_payload(self, message, signer_public_key):
+        """Make a payload for the given message type"""
+        if isinstance(message, role_transaction_pb2.CreateRole):
+            message_type = RBACPayload.CREATE_ROLE
+        elif isinstance(message, role_transaction_pb2.ProposeAddRoleMember):
+            message_type = RBACPayload.PROPOSE_ADD_ROLE_MEMBERS
+        elif isinstance(message, role_transaction_pb2.ProposeAddRoleOwner):
+            message_type = RBACPayload.PROPOSE_ADD_ROLE_OWNERS
+        elif isinstance(message, role_transaction_pb2.ProposeAddRoleAdmin):
+            message_type = RBACPayload.PROPOSE_ADD_ROLE_ADMINS
+        else:
             raise TypeError(
-                "Expected role to be a role_transaction_pb2.CreateRole, use make first"
+                "RoleManager.make_payload doesn't support message type {}".format(
+                    type(message)
+                )
             )
 
-        inputs = [
-            addresser.make_sysadmin_members_address(signer_keypair.public_key),
-            addresser.make_role_attributes_address(role.role_id),
-        ]
-        inputs.extend([addresser.make_user_address(u) for u in role.admins])
-        inputs.extend([addresser.make_user_address(u) for u in role.owners])
-        inputs.extend(
-            [
-                addresser.make_role_admins_address(role_id=role.role_id, user_id=a)
-                for a in role.admins
-            ]
+        inputs, outputs = self.make_addresses(
+            message=message, signer_public_key=signer_public_key
         )
-        inputs.extend(
-            [
-                addresser.make_role_owners_address(role_id=role.role_id, user_id=o)
-                for o in role.owners
-            ]
+        return self.batch.make_payload(
+            message=message, message_type=message_type, inputs=inputs, outputs=outputs
         )
 
-        transaction = self.batch.make_transaction(
-            message=role,
-            message_type=RBACPayload.CREATE_ROLE,
-            inputs=inputs,
-            outputs=inputs,
-            signer_keypair=signer_keypair,
-        )
+    def make_addresses(self, message, signer_public_key=None):
+        """Makes the approporiate inputs & output addresses for the message type"""
+        if isinstance(message, role_transaction_pb2.CreateRole):
+            inputs = [
+                # addresser.make_sysadmin_members_address(signer_public_key),
+                addresser.make_role_attributes_address(message.role_id)
+            ]
+            inputs.extend([addresser.make_user_address(u) for u in message.admins])
+            inputs.extend([addresser.make_user_address(u) for u in message.owners])
+            inputs.extend(
+                [
```

Comment:
Has python formatter been run on this?  The formatting seems odd.

Reply:
Yes, formatter has been run.